app/src/marketplace/components/StreamPreviewPage/StreamLivePreview/index.jsx

```diff
@@ -30,160 +34,70 @@ export type DataPoint = {
-                },
-            }, (data, metadata) => this.onData({
-                data,
-                metadata,
-            }))
-        } catch (e) {
-            this.setState({
-                subscriptionError: true,
-            })
-        }
-
-        if (this.subscription) {
-            // Log errors thrown by stream
-            this.subscription.on('error', () => {
-                if (!this.state.dataError) {
-                    this.setState({
-                        dataError: true,
-                    })
-                }
-            })
+const prettyPrintData = (data: ?{}, compact: boolean = false) => stringifyObject(data, {
+    indent: '  ',
+    inlineCharacterLimit: compact ? Infinity : 5,
+})
+
+const StreamLivePreview = ({
+    streamId,
+    selectedDataPoint,
+    onSelectDataPoint,
+    userpagesPreview,
+    hasData,
+    run = true,
+}: Props) => {
+    const dataRef = useRef([])
+    const [visibleData, setVisibleData] = useState([])
+    const [dataError, setDataError] = useState(false)
+    const [mobileTableColumnIndex, setMobileTableColumnIndex] = useState(0)
+    const { hasLoaded, client } = useContext(ClientContext)
+    const isMounted = useIsMounted()
+
+    const updateDataToState = useCallback(throttle((data) => {
+        if (hasData && visibleData.length === 0) {
+            hasData()
         }
-    }
+        setVisibleData([...data])
 
-    unsubscribe = () => {
-        if (this.client && this.subscription) {
-            this.client.unsubscribe(this.subscription)
+        if (!selectedDataPoint && data.length) {
+            onSelectDataPoint(data[0], true)
         }
-    }
-
-    prettyPrintData = (data: ?{}, compact: boolean = false) => stringifyObject(data, {
-        indent: '  ',
-        inlineCharacterLimit: compact ? Infinity : 5,
-    })
+    }, 100), [hasData, selectedDataPoint])
 
-    subscription: any
-    data: Array<{}> = []
-    client: StreamrClient
+    const onData = useCallback((data, metadata) => {
+        if (!isMounted()) { return }
 
-    updateDataToState = throttle((data) => {
-        this.setState({
-            visibleData: [...data],
-        })
-        if (!this.props.selectedDataPoint && data.length) {
-            this.props.onSelectDataPoint(data[0], true)
+        const dataPoint: DataPoint = {
+            data,
+            metadata,
         }
-    }, 100)
 
-    clearData = () => {
-        this.data = []
-        this.setState({
-            visibleData: [],
-        })
-    }
+        dataRef.current.unshift(dataPoint)
+        dataRef.current.length = Math.min(dataRef.current.length, LOCAL_DATA_LIST_LENGTH)
+        updateDataToState(dataRef.current)
+    }, [dataRef, updateDataToState, isMounted])
 
-    render() {
-        const { userpagesPreview } = this.props
-        const tz = moment.tz.guess()
-        const { visibleData, mobileTableColumnIndex } = this.state
-        return (
+    return (
+        <SubscriptionStatusProvider>
```

Comment:
Does this use any of the information from `SubscriptionStatusProvider`? This thing only exists because canvases need to know when all the subscriptions are ready before starting the canvas.

Reply:
No, it's not needed - just added as a wrapper instead of `Fragment`. Can be safely removed.